libs/langgraph/langgraph/prebuilt/sequential_executor.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Optional, Type, Union, cast
+
+from langchain_core.runnables.base import Runnable, RunnableLike
+
+from langgraph.checkpoint.base import BaseCheckpointSaver
+from langgraph.graph.state import END, START, CompiledStateGraph, StateGraph
+from langgraph.store.base import BaseStore
+
+
+def _get_name(step: RunnableLike) -> str:
+    if isinstance(step, Runnable):
+        if step.name is None:
+            raise ValueError(
+                f"Runnable ({step}) needs to have a name attribute. "
+                "Consider setting the name or passing it as a tuple (name, runnable)."
+            )
+        return step.name
+    elif callable(step):
+        return getattr(step, "__name__", step.__class__.__name__)
+    else:
+        raise TypeError(f"Unsupported step type: {step}")
+
+
+def create_sequential_executor(
+    *steps: Union[RunnableLike, tuple[str, RunnableLike]],
+    state_schema: Type[Any],
+    checkpointer: Optional[BaseCheckpointSaver] = None,
+    store: Optional[BaseStore] = None,
+    interrupt_before: Optional[list[str]] = None,
+    interrupt_after: Optional[list[str]] = None,
+    debug: bool = False,
+) -> CompiledStateGraph:
+    """Creates a sequential executor graph that runs a series of provided steps in order.
+
+    Args:
+        *steps: A sequence of RunnableLike objects or (name, RunnableLike) tuples.
```

Comment:
* what happens if the same name is re-used by mistake?
* What happens if the same name is re-used on purpose? (e.g., we want to make a sequence that re-uses the same node?)

Reply:
> what happens if the same name is re-used by mistake?

langgraph already handles this at graph.compile()

> What happens if the same name is re-used on purpose? (e.g., we want to make a sequence that re-uses the same node?)

it would have to have a different name, we don't allow multiple nodes with the same name in langgraph

Reply:
added a check for the duplicate names actually!